src/float.jl

```diff
@@ -0,0 +1,147 @@
+module FloatEncoding
+using Supposition: uint, tear, bias, fracsize, exposize, max_exponent, assemble
+
+"""
+    exponent_key(T, e)
+
+A lexographical ordering for floating point exponents. The encoding is taken
+from hypothesis.
+The ordering is
+- non-negative exponents in increasing order
+- negative exponents in decreasing order
+- the maximum exponent
+"""
+function exponent_key(::Type{T}, e::iT) where {T<:Base.IEEEFloat,iT<:Unsigned}
+    if e == max_exponent(T)
+        return Inf
+    end
+    unbiased = float(e) - bias(T)
+    if unbiased < 0
+        10000 - unbiased
```

Comment:
10000 is simply greater than key assigned to any positive exponent so that negative exponents are ordered after positive exponents. In principle we could use something like `bias(T)+1`.